### PR TITLE
Add CMake build

### PR DESCRIPTION
**Pull request by mcm001**

### CODE REVIEW

File: common/system/states/cli/state_cli_offload.cpp [hunk 1]
```diff
@@ -91,8 +91,7 @@ EndCondition_e CliOffloadState::run() {
         flight_timestamp = (time_t)metadataPacket->gpsTimestamp;
         // Check timestamp between 2000 and 2100
         if (flight_timestamp > 946702800 && flight_timestamp < 4102462800) {
-          snprintf(timeString, sizeof(timeString), "%" PRIu64,
-                   flight_timestamp);
+          snprintf(timeString, sizeof(timeString), "%llu", flight_timestamp);
```
[the remark this answers is not in the document]
Comment (mcm001) [reply common/system/states/cli/state_cli_offload.cpp:94]: My compiler is still upset even with these changes so I'll go with no

File: .vscode/launch.json [hunk 2]
```diff
@@ -4,44 +4,178 @@
...
+          }
+        ]
+      },
+      "preLaunchTask": "Build amazon-stm32-arduino"
+    },
+    {
+      "type": "platformio-debug",
+      "request": "launch",
+      "name": "PIO Debug",
+      "executable": "/home/matt/Documents/GitHub/stm32-avionics/.pio/build/blink/firmware.elf",
```
[the remark this answers is not in the document]
Comment (mcm001) [reply .vscode/launch.json:149]: That's shit that platformio changes every single time you open the project in VSCode. I only actually added the cortex-debug tasks

File: cmake/stm32/f0.cmake
[the remark this answers is not in the document]
Comment (SamHaggans) [reply cmake/stm32/f0.cmake:1]: Can you just use `--recurse-submodules`?

File: .vscode/launch.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (SamHaggans) [reply .vscode/launch.json]: Oh alright, makes sense

File: cmake/stm32/f0.cmake
[the remark this answers is not in the document]
Comment (mcm001) [reply cmake/stm32/f0.cmake:1]: Yeah, but then you can't just blindly `git clone`, you need to add `--recurse-submodules` for one dumb cmake toolchain thing

**Review by SamHaggans — APPROVED** [review]

**Pull request merged**